File: app/src/main/java/com/abdelmeged/ahmed/networkitotest/java/NormalActivity2.java
package com.abdelmeged.ahmed.networkitotest.java;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.widget.Toast;

import com.abdelmeged.ahmed.networkitotest.R;
import com.ahmedabdelmeged.networkito.ConnectionType;
import com.ahmedabdelmeged.networkito.ConnectivityChangeListener;
import com.ahmedabdelmeged.networkito.Networkito;


public class NormalActivity2 extends AppCompatActivity implements ConnectivityChangeListener {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Networkito networkito = new Networkito(this, this, this);
    }

    @Override
    public void onInternetConnected(ConnectionType connectionType) {
        Toast.makeText(this, "connected: " + connectionType.name(), Toast.LENGTH_SHORT).show();
    }

    @Override
    public void onInternetDisconnected() {
        Toast.makeText(this, "disconnected", Toast.LENGTH_SHORT).show();
    }

}

File: app/src/main/java/com/abdelmeged/ahmed/networkitotest/kotlin/NormalActivity2.kt
package com.abdelmeged.ahmed.networkitotest.kotlin

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import com.abdelmeged.ahmed.networkito.ConnectionType
import com.abdelmeged.ahmed.networkito.ConnectivityChangeListener
import com.abdelmeged.ahmed.networkito.Networkito
import com.abdelmeged.ahmed.networkitotest.R
import org.jetbrains.anko.toast

class NormalActivity2 : AppCompatActivity(), ConnectivityChangeListener {

    lateinit var networkito: Networkito

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        networkito = Networkito(this, this)
    }

    override fun onInternetConnected(connectionType: ConnectionType?) {
        toast("connected: " + connectionType!!.name)
    }

    override fun onInternetDisconnected() {
        toast("disconnected")
    }

    override fun onStop() {
        super.onStop()
        networkito.unRegisterNetworkitoReceiver()
    }
}
